File: app/src/main/java/com/example/bloodbank/ui/fragment/authCycle/RestPasswordFragment.java
package com.example.bloodbank.ui.fragment.authCycle;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;

import com.example.bloodbank.R;
import com.example.bloodbank.data.local.SharedPreferencesManger;
import com.example.bloodbank.data.model.restPassword.RestPassword;
import com.example.bloodbank.ui.fragment.BaseFragment;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static com.example.bloodbank.data.api.ApiClient.getClient;
import static com.example.bloodbank.data.local.SharedPreferencesManger.PHONE;
import static com.example.bloodbank.data.local.SharedPreferencesManger.SaveData;
import static com.example.bloodbank.data.local.SharedPreferencesManger.setSharedPreferences;
import static com.example.bloodbank.helper.HelperMethods.dismissProgressDialog;
import static com.example.bloodbank.helper.HelperMethods.replace;
import static com.example.bloodbank.helper.HelperMethods.showProgressDialog;
import static com.example.bloodbank.helper.HelperMethods.showToast;

public class RestPasswordFragment extends BaseFragment {

    @BindView(R.id.fragment_rest_password_ed_phone)
    EditText fragmentForgetPasswordEdPhone;
    private NewPasswordFragment newPasswordFragment;
    private static final String TAG = "RestPasswordFragment";

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        setupAcvtivity();
        View view = inflater.inflate(R.layout.fragment_rest_password, container, false);
        ButterKnife.bind(this, view);
        setSharedPreferences(getActivity());
        return view;
    }

    private void restPassword(String phone) {
        newPasswordFragment = new NewPasswordFragment();
        showProgressDialog(getActivity(), getString(R.string.please_wait));
        getClient().resetPassword(phone).enqueue(new Callback<RestPassword>() {
            @Override
            public void onResponse(Call<RestPassword> call, Response<RestPassword> response) {
                dismissProgressDialog();
                if (response.body().getStatus() == 1) {
                    showToast(getActivity(), response.body().getMsg());
                    SaveData(getActivity(), PHONE, phone);
                    replace(newPasswordFragment, getFragmentManager(), R.id.activity_auth_fl_content);
                } else {
                    showToast(getActivity(), response.body().getMsg());
                }
            }

            @Override
            public void onFailure(Call<RestPassword> call, Throwable t) {

            }
        });
    }

    @OnClick(R.id.fragment_rest_password_btn_send)
    void onViewClicked() {
        restPassword(fragmentForgetPasswordEdPhone.getText().toString().trim());
    }

    @Override
    public void onBack() {
        LoginFragment loginFragment = new LoginFragment();
        replace(loginFragment, getActivity().getSupportFragmentManager(), R.id.activity_auth_fl_content);
    }
}

File: app/src/main/java/com/example/bloodbank/ui/fragment/authCycle/RestPasswordFragment.kt
package com.example.bloodbank.ui.fragment.authCycle

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.bloodbank.R
import com.example.bloodbank.data.local.SharedPreferencesManger.setSharedPreferences
import com.example.bloodbank.databinding.FragmentRestPasswordBinding
import com.example.bloodbank.extensions.inflateWithBinding
import com.example.bloodbank.extensions.replaceFragment
import com.example.bloodbank.ui.fragment.BaseFragment
import com.example.bloodbank.ui.viewModels.AuthViewModel
import org.koin.android.ext.android.inject

class RestPasswordFragment : BaseFragment(), View.OnClickListener {

    lateinit var binding: FragmentRestPasswordBinding
    private val viewModel by inject<AuthViewModel>()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        setupActivity()
        binding = inflater.inflateWithBinding(R.layout.fragment_rest_password, container)

        setSharedPreferences(activity!!)
        binding.fragmentRestPasswordBtnSend.setOnClickListener(this)
        return binding.root
    }

    override fun onClick(v: View?) {
        viewModel.restPassword(binding.fragmentRestPasswordEdPhone.text.toString().trim(' '), this.activity!!)
    }

    override fun onBack() {
        activity.replaceFragment(R.id.activity_auth_fl_content, LoginFragment())
    }
}